oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/ModelService.kt

```diff
@@ -0,0 +1,68 @@
+package com.empowerops.volition.ref_oasis
+
+import com.empowerops.volition.dto.ErrorResponseDTO
+import com.empowerops.volition.dto.NodeStatusCommandOrResponseDTO
+import com.empowerops.volition.dto.OASISQueryDTO
+import com.empowerops.volition.dto.SimulationResponseDTO
+import io.grpc.stub.StreamObserver
+import javafx.beans.property.SimpleStringProperty
+import javafx.collections.ObservableList
+import kotlinx.coroutines.channels.Channel
+import java.time.Duration
+import java.time.LocalDateTime
+
+interface ModelService {
+    var simByName: Map<String, Simulation>
```

Comment:
given that you have `addSim`, `updateSim`, etc, can you remove this map?

Reply:
As an alternative exercise, can you remove the other methods, and make this code still work?

Hint: if you change this to `val simByName: MutableMap<String, Simulation>`, then you can create your own object that implements `MutableMap`....

Reply:
I didn't address these but I will try to take another look in next issue